client/state/data-layer/wpcom/read/recommendations/sites/index.js

```diff
@@ -44,22 +42,21 @@ export const fromApi = response => {
 	} ) );
```

Comment:
doodling…

```js
export const fromApi = ( { algorithm, sites } ) => sites.map( site => ( {
	feedId: site.feed_id,
	blogId: site.blog_id,
	title: decodeEntities( site.blog_title ),
	url: site.blog_url,
	railcar: site.railcar,
	algorithm,
} ) );
```

why? the `if ( ! response )` shouldn't be needed. if the data we expect is missing from the API response can we really do anything with it? won't it be a failed request? if so, then just access `response.sites.map` which will throw a `TypeError` if the data is missing and the system will auto-switch to `onError`

now the `onSuccess` function doesn't take `( { sites } )` but rather just `( sites )`

Reply:
I'm always down for simplifying code. Currently we have a test which ensures we at least return an empty Array in case of an error:

https://github.com/Automattic/wp-calypso/blob/9366ed5ef3dca679e4eed31641a232e989064470/client/state/data-layer/wpcom/read/recommendations/sites/test/index.js#L74-L78

We can remove this test if propagating an error is what we want. I think, however, we need to define behaviour for what happens in that case because right now it's a noop:

https://github.com/Automattic/wp-calypso/blob/9366ed5ef3dca679e4eed31641a232e989064470/client/state/data-layer/wpcom/read/recommendations/sites/index.js#L58

UI-wise this results in a never ending placeholder. The same happens already if we can't fetch recommendations due to e.g. network issues (and return an empty array from `fromApi`); because of that I'd be fine to add your suggested changes and think about what happens in an error case in a later iteration.

Reply:
Addressed in #27954